File: fizzbuzz.hs
module FizzBuzz(fizzbuzz) where

fizzbuzz :: Int -> [String]
fizzbuzz n = map fizzbuzz' [1..n]
  where fizzbuzz' el | el `rem` 15 == 0 = "FizzBuzz"
                     | el `rem` 5 == 0 = "Buzz"
                     | el `rem` 3 == 0 = "Fizz"
                     | otherwise = show el

File: fizzbuzz.ex
defmodule FizzBuzz do
  def val(n) when rem(n, 15) == 0, do: "FizzBuzz"
  def val(n) when rem(n, 3) == 0, do: "Fizz"
  def val(n) when rem(n, 5) == 0, do: "Buzz"
  def val(n), do: n

  def fizzbuzz(n) do
    Enum.map(1..n, fn(x) -> val(x) end)
  end
end